src/LtiMessageLaunch.php

```diff
@@ -147,6 +173,32 @@ public function validate(array $request = null)
             ->cacheLaunchData();
     }
 
+    public function migrate()
+    {
+        if (isset($this->deployment) || !$this->canMigrate()) {
```

Comment:
Let me describe a situation here.  A user used 1.1 and wants to migrate. Their LMS however does not support the 1p1 migrate claim or the oauth_consumer_key_sign. Despite this however, using things like the `toolPlatform->guid` claim, we can reasonably infer that this user NEEDS to migrate, even though they didn't pass the `oauth_consumer_key`. In that case, I, the tool provider, need a different error out of packback. They "can't" migrate, but they "should" have migrated. This helps me in the messaging I will show them, as well as handling manually creating the deployment, to ensure the re-binding can occur manually on my side. This migrate function should change slightly.

1. The `canMigrate` function could be renamed to `implementsMigrationDb()` and check if the tool provider implemented code for migration, and if findLti1p1Keys returns any keys. If the deployment is set, or they didn't implement migration, or there is no key to migrate, then just return, as migration either isn't necessary, or isn't supported.
2. If we found a key, but the system didn't pass the key_sign or the consumer_key, then we shuld throw an error saying the KEY_SIGN or KEY is missing.
3. If we have the keys, but the signatures don't match, we should throw a KEY_SIGN_MISMATCH error.
4. Otherwise, we should do the migration.

Reply:
Let me make sure I understand the scenario you described in the first paragraph:

1. A user performs an LTI 1.3 launch.
2. The LMS does not support migrations, and therefor the `oauth_consumer_key` is not included in the launch.
3. Despite this, there is a matching LTI 1.1 key that could be migrated (i.e., calling `findMatchingLti1p1Keys()` returns several results), but cannot be automatically due to the missing claims.

In that case, how would you know whether a user has already migrated or not? What in the system indicates "the user should migrate" vs "the user already migrated"?